app/controllers/users_controller.rb

```diff
@@ -1,14 +1,21 @@
 get '/users/new' do
+  @user = User.new(user: params[:username], password: params[:password_hash])
   erb :'users/new'
 end
 
 post '/users' do
-  user = User.new(params)
-  if user.save
-    redirect '/'
+  @user = User.new(params[:user])
```

Comment:
since you arent passing user anywhere from this post no need for it to be an instance variable